File: Python/8kyu/8kyu-opposites-attract.py
# Opposites Attract
# Language: Python
# Difficulty: 8 kyu

# Solution 1
def lovefunc( flower1, flower2 ):
    return flower1 % 2 != flower2 % 2

# Refactored Solution 1
def lovefunc( flower1, flower2 ):
    return (flower1 + flower2) % 2


# Tests
import codewars_test as test
from solution import lovefunc

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(lovefunc(1,4), True)
        test.assert_equals(lovefunc(2,2), False)
        test.assert_equals(lovefunc(0,1), True)
        test.assert_equals(lovefunc(0,0), False)

File: R/8kyu/8kyu-opposites-attract.r
# Opposites Attract
# Language: R
# Difficulty: 8 kyu

# Solution 1
lovefunc <- function(flower1, flower2){
  ifelse(flower1 %% 2 == flower2 %% 2, FALSE, TRUE)
}

# Refactored Solution 1
lovefunc <- function(flower1, flower2){
  flower1 %% 2 != flower2 %% 2
}

# Tests
test_that('Basic tests', {
  expect_equal(lovefunc(1,4), TRUE)
  expect_equal(lovefunc(2,2), FALSE)
  expect_equal(lovefunc(0,1), TRUE)
  expect_equal(lovefunc(0,0), FALSE)
})